server/src/main/java/com/defold/extender/Extender.java

```diff
@@ -760,6 +760,14 @@ else if (platform.contains("osx")) {
         Map<String, Object> mergedContextWithPodsForObjC = ExtenderUtil.mergeContexts(trimmedContext, podContextObjC);
         Map<String, Object> mergedContextWithPodsForObjCpp = ExtenderUtil.mergeContexts(trimmedContext, podContextObjCpp);
 
+        // remove systemIncludes from objc and objc++
+        // this is a bit crude but cocoapod builds do not provide any -isystem option and
+        // it seems like the "{{env.SYSROOT}}/usr/include/c++/v1" set in build.yml is
+        // causing problems when building objc code with -fmodules enabled
```

Comment:
Yes, we currently don't separate the C and C++ languages.
And C++ headers cannot be used in C languages (e.g. ObjC)